demo/state.h

```diff
@@ -78,6 +98,52 @@ struct State {
                                   HTTPResponseCode::OK,
                                   "image/svg+xml");
   }
+
+  void DemoRequest(Request r) {
+    if (r.http.Method() == "POST") {
+      // TODO(dkorolev): This should get simpler once Bricks 1.0 is out, the `.http.` will go away.
+      if (!r.http.HasBody()) {
+        points.emplace_back(atof(r.url.query["x"].c_str()),
+                            atof(r.url.query["y"].c_str()),
+                            !!atoi(r.url.query["label"].c_str()));
+      } else {
+        // Note: this will throw on an invalid JSON. TODO(dkorolev): Think of it further :-)
```

Comment:
Throws in the request handling code should result in a `HTTP 500` response and an error log record (and this behavior should be overridable with custom error handling).